pkg/mcp/server/server.go

```diff
@@ -87,6 +87,11 @@ type Server struct {
 // this case and apply appropriate policy.
 type AuthChecker interface {
 	Check(authInfo credentials.AuthInfo) error
+	// ShouldRecordFailure is to rate limit auth check failures
+	// to avoid flooding logs with duplicate information.
+	// it returns with bool and int value that indicate failure counts
+	// since last time.
+	ShouldRecordFailure() (bool, int)
```

Comment:
I don't think we should add this method to the interface. It should not be the responsibility of the AuthChecker to keep track of failures. The server object itself is probably a better place to do this.

Reply:
Ack